// some-random-api/dist/structures/OtherCategory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.OtherCategory = void 0;
const Category_1 = require("./Category");
const Errors_1 = require("../util/Errors");
class OtherCategory extends Category_1.Category {
    chatbot(message) {
        if (!message)
            throw new Errors_1.MissingParameter('message');
        return this.request('/chatbot', { message });
    }
    mc(username) {
        if (!username)
            throw new Errors_1.MissingParameter('username');
        return this.request('/mc', { username });
    }
    lyrics(title) {
        if (!title)
            throw new Errors_1.MissingParameter('title');
        return this.request('/lyrics', { title });
    }
    binaryEncode(text) {
        if (!text)
            throw new Errors_1.MissingParameter('text');
        return this.request('/binary', { text });
    }
    binaryDecode(decode) {
        if (!decode)
            throw new Errors_1.MissingParameter('decode');
        return this.request('/binary', { decode });
    }
    base64Encode(encode) {
        if (!encode)
            throw new Errors_1.MissingParameter('encode');
        return this.request('/base64', { encode });
    }
    base64Decode(decode) {
        if (!decode)
            throw new Errors_1.MissingParameter('decode');
        return this.request('/base64', { decode });
    }
    meme() {
        return this.request('/meme');
    }
}
exports.OtherCategory = OtherCategory;
