// some-random-api/dist/structures/WelcomeCategory.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.WelcomeCategory = void 0;
const Category_1 = require("./Category");
class WelcomeCategory extends Category_1.Category {
    constructor() {
        super('', 'https://welcome-imgs.some-random-api.ml');
    }
    join(options) {
        return this.requestBuffer(`/img/${options.template}/${options.background}`, {
            type: 'join',
            username: options.tag.split('#')[0],
            discriminator: options.tag.split('#')[1],
            guildName: options.guildName,
            memberCount: options.memberCount,
            avatar: options.avatar,
            textcolor: options.textcolor
        });
    }
    leave(options) {
        return this.requestBuffer(`/img/${options.template}/${options.background}`, {
            type: 'leave',
            username: options.tag.split('#')[0],
            discriminator: options.tag.split('#')[1],
            guildName: options.guildName,
            memberCount: options.memberCount,
            avatar: options.avatar,
            textcolor: options.textcolor
        });
    }
}
exports.WelcomeCategory = WelcomeCategory;
